config/mq/config.go

```diff
@@ -50,4 +50,11 @@ func Configure(p *config.Provider) {
 			return conn, nil
 		}
 	})
+
+	p.AddResourceConfigurator("aws_mq_user", func(r *config.Resource) {
+		r.References["broker_id"] = config.Reference{
+			Type: "Broker",
```

Comment:
This way the cross-resource reference is generated with a target resource API version set to the source's version, i.e., `v1alpha1`. However, `Broker.v1alpha1.mq` does not exist. We have deprecated `config.Resource.Type` for reasons similar to this in favor of the higher level API `config.Reference.TerraformName`, which automatically handles resolution of the correct kind & version for reference target:
```suggestion
			TerraformName: "aws_mq_broker",
```